src/docker/install.ts

```diff
@@ -127,12 +134,22 @@ export class Install {
         const cli = await HubRepository.build('dockereng/cli-bin');
         extractFolder = await cli.extractImage(tag);
 
+        const moby = await HubRepository.build('moby/moby-bin');
         if (['win32', 'linux'].includes(platform)) {
           core.info(`Downloading dockerd from moby/moby-bin:${tag}`);
-          const moby = await HubRepository.build('moby/moby-bin');
           await moby.extractImage(tag, extractFolder);
         } else if (platform == 'darwin') {
-          // On macOS, the docker daemon binary will be downloaded inside the lima VM
+          // On macOS, the docker daemon binary will be downloaded inside the lima VM.
+          // However, we will get the exact git revision from the image config
+          // to get the matching systemd unit files.
+          const manifest = await moby.getPlatformManifest(tag);
+          const config = await moby.getConfig<ImageConfig>(manifest.config.digest);
```

Comment:
I think labels and annotations will only by on the manifest list config

Reply:
Looking at `docker buildx imagetools inspect  moby/moby-bin:latest --raw `, index doesn't have these:
```json
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.index.v1+json",
  "manifests": [
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:c52ff4ad684b729f597eefacf1abc65913dcb3423071b71997cd271690c8a9b7",
      "size": 1630,
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:fda20c7387897ee54811fd74b96472c4a454a62f29b5eb595caaeb4e04f01bd5",
      "size": 567,
      "annotations": {
        "vnd.docker.reference.digest": "sha256:c52ff4ad684b729f597eefacf1abc65913dcb3423071b71997cd271690c8a9b7",
        "vnd.docker.reference.type": "attestation-manifest"
      },
      "platform": {
        "architecture": "unknown",
        "os": "unknown"
      }
...
```

Reply:
The image config has them though:
```json
docker buildx imagetools inspect  moby/moby-bin:latest@sha256:fe77a2f844e3db621de180b88e0e8c5439dd6802e4e6efcc5371be4a05e252d3 --raw | jq
{
  "architecture": "amd64",
  "config": {
    "Env": [
      "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
    ],
    "WorkingDir": "/",
    "Labels": {
      "org.opencontainers.image.created": "2024-09-20T18:10:20.768Z",
      "org.opencontainers.image.description": "The Moby Project - a collaborative project for the container ecosystem to assemble container-based systems",
      "org.opencontainers.image.licenses": "Apache-2.0",
      "org.opencontainers.image.revision": "41ca978a0a5400cc24b274137efa9f25517fcc0b",
      "org.opencontainers.image.source": "https://github.com/moby/moby",
      "org.opencontainers.image.title": "moby",
      "org.opencontainers.image.url": "https://github.com/moby/moby",
      "org.opencontainers.image.version": "27.3.1"
    }
  },
```

Reply:
Good catch